### ISSUE TITLE
Cant create new argocd application by using argocd cli

### ISSUE THREAD

**Issue by imrenagi** [argoproj/argo-cd#4573]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

when run `argocd app create`, we got this error:
```
FATA[0000] rpc error: code = Unknown desc = Post "https://example.com:443/application.ApplicationService/Create": EOF
```

However when we run the other command like
```
$ argocd cluster list ....
$ argocd login argocd.example.com --sso
```
This command works as expected

**To Reproduce**

we run 
```
argocd app create guestbook1 --repo https://github.com/example/podinfo --path path-to-app --dest-namespace default --dest-server https://kubernetes.default.svc --directory-recurse
```
it returns
```
FATA[0000] rpc error: code = Unavailable desc = transport is closing
```

When we run this `argocd app create` with `--grpc-web` :
```
argocd app create guestbook1 --repo https://github.com/example/podinfo --path path-to-app --dest-namespace default --dest-server https://kubernetes.default.svc --directory-recurse --grpc-web
```
we got this following error:
```
FATA[0000] rpc error: code = Unknown desc = Post "https://example.com:443/application.ApplicationService/Create": EOF
```

**Expected behavior**

`argocd app create` should be able to add new application

**Version**

```shell
argocd: v1.7.6+b04c25e
  BuildDate: 2020-09-19T00:53:13Z
  GitCommit: b04c25eca8f1660359e325acd4be5338719e59a0
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.7.6+b04c25e
  BuildDate: 2020-09-19T00:52:04Z
  GitCommit: b04c25eca8f1660359e325acd4be5338719e59a0
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: {Version:kustomize/v3.6.1 GitCommit:c97fa946d576eb6ed559f17f2ac43b3b5a8d5dbd BuildDate:2020-05-27T20:47:35Z GoOs:linux GoArch:amd64}
  Helm Version: version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
  Kubectl Version: v1.17.8
```

**Logs**

```
 {"level":"info","msg":"Initializing OIDC provider (issuer: https://example.com/api/dex)","time":"2020-10-15T10:31:26Z"}
 {"level":"info","msg":"OIDC supported scopes: [openid email groups profile offline_access]","time":"2020-10-15T10:31:26Z"}
 2020/10/15 10:31:26 proto: tag has too few fields: "-"
 panic: interface conversion: *time.Time is not proto.Message: missing method ProtoMessage [recovered]
    panic: interface conversion: *time.Time is not proto.Message: missing method ProtoMessage

 goroutine 1049 [running]:
 encoding/json.(*encodeState).marshal.func1(0xc001fb0ad0)
    /usr/local/go/src/encoding/json/encode.go:326 +0x9a
 panic(0x1f2e740, 0xc000e2a7e0)
    /usr/local/go/src/runtime/panic.go:967 +0x166
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalValue(0x3a40ec0, 0xc001fb06d0, 0xc001d46540, 0x21b43a0, 0xc001fa8088, 0x199, 0x0, 0x0, 0x1, 0xc001d46540)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:567 +0x1d91
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalField(0x3a40ec0, 0xc001fb06d0, 0xc001d46540, 0x21b43a0, 0xc001fa8088, 0x199, 0x0, 0x0, 0x0, 0xc000a044f8)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:488 +0xe5
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalObject(0x3a40ec0, 0xc001fb06d0, 0x7fa473d32eb8, 0xc001fa8088, 0x0, 0x0, 0x0, 0x0, 0x220d741, 0x1)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:346 +0x4d5
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalValue(0x3a40ec0, 0xc001fb06d0, 0xc001d46480, 0x21b1200, 0xc001fa8088, 0x199, 0x0, 0x0, 0x0, 0x0)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:567 +0x1e44
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalField(0x3a40ec0, 0xc001fb06d0, 0xc001d46480, 0x21b1200, 0xc001fa8088, 0x199, 0x0, 0x0, 0x68, 0xc000a044d0)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:488 +0xe5
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalObject(0x3a40ec0, 0xc001fb06d0, 0x7fa473d32e08, 0xc001fa8020, 0x0, 0x0, 0x0, 0x0, 0x220d741, 0x1)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:346 +0x4d5
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalValue(0x3a40ec0, 0xc001fb06d0, 0xc001d46300, 0x2177380, 0xc001fa8020, 0x199, 0x0, 0x0, 0x0, 0x0)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:567 +0x1e44
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalField(0x3a40ec0, 0xc001fb06d0, 0xc001d46300, 0x2177380, 0xc001fa8020, 0x199, 0x0, 0x0, 0x20, 0xc000a04430)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:488 +0xe5
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalObject(0x3a40ec0, 0xc001fb06d0, 0x2665c80, 0xc001fa8000, 0x0, 0x0, 0x0, 0x0, 0x220d741, 0x1)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:346 +0x4d5
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalValue(0x3a40ec0, 0xc001fb06d0, 0xc001d46180, 0x207d3a0, 0xc001fa8000, 0x199, 0x0, 0x0, 0x1, 0xc001d46180)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:567 +0x1e44
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalField(0x3a40ec0, 0xc001fb06d0, 0xc001d46180, 0x207d3a0, 0xc001fa8000, 0x199, 0x0, 0x0, 0x0, 0xc000a043c0)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:488 +0xe5
 github.com/golang/protobuf/jsonpb.(*Marshaler).marshalObject(0x3a40ec0, 0xc001fb06d0, 0x2663a80, 0xc001fa8000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1d1edb2)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:346 +0x4d5
 github.com/golang/protobuf/jsonpb.(*Marshaler).Marshal(0x3a40ec0, 0x2629b20, 0xc000e2a780, 0x2663a80, 0xc001fa8000, 0x40a6f3, 0x1f316c0)
    /go/pkg/mod/github.com/golang/protobuf@v1.3.2/jsonpb/jsonpb.go:139 +0x1a6
 github.com/argoproj/argo-cd/util/grpc.(*jsonpbMarshalleble).MarshalJSON(0xc00018a070, 0x1f93720, 0xc00018a070, 0x7fa473d32c98, 0xc00018a070, 0xc00018a001)
    /go/src/github.com/argoproj/argo-cd/util/grpc/logging.go:35 +0x87
 encoding/json.marshalerEncoder(0xc0013d4480, 0x1f93720, 0xc00018a070, 0x16, 0xc000180100)
    /usr/local/go/src/encoding/json/encode.go:477 +0xba
 encoding/json.(*encodeState).reflectValue(0xc0013d4480, 0x1f93720, 0xc00018a070, 0x16, 0xc000180100)
    /usr/local/go/src/encoding/json/encode.go:358 +0x82
 encoding/json.interfaceEncoder(0xc0013d4480, 0x1ec1000, 0xc001396190, 0x94, 0xc001390100)
    /usr/local/go/src/encoding/json/encode.go:713 +0xac
 encoding/json.mapEncoder.encode(0x2371c70, 0xc0013d4480, 0x1f36940, 0xc000e2a6c0, 0x15, 0x1f30100)
    /usr/local/go/src/encoding/json/encode.go:800 +0x354
 encoding/json.(*encodeState).reflectValue(0xc0013d4480, 0x1f36940, 0xc000e2a6c0, 0x15, 0xc001fb0100)
    /usr/local/go/src/encoding/json/encode.go:358 +0x82
 encoding/json.(*encodeState).marshal(0xc0013d4480, 0x1f36940, 0xc000e2a6c0, 0xc000a70100, 0x0, 0x0)
    /usr/local/go/src/encoding/json/encode.go:330 +0xf0
 encoding/json.(*Encoder).Encode(0xc001fb0c78, 0x1f36940, 0xc000e2a6c0, 0x5, 0xc000b04bf8)
    /usr/local/go/src/encoding/json/stream.go:206 +0x8a
 github.com/sirupsen/logrus.(*JSONFormatter).Format(0xc000b8cec0, 0xc001d641c0, 0xc001fb0e48, 0x534dbd, 0x1ff3da0, 0xc000e2a690, 0xc001fb0e90)
    /go/pkg/mod/github.com/sirupsen/logrus@v1.6.0/json_formatter.go:120 +0x63d
 github.com/sirupsen/logrus.(*Entry).write(0xc001d641c0)
    /go/pkg/mod/github.com/sirupsen/logrus@v1.6.0/entry.go:275 +0x7c
 github.com/sirupsen/logrus.Entry.log(0xc000138000, 0xc000e2a660, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    /go/pkg/mod/github.com/sirupsen/logrus@v1.6.0/entry.go:251 +0x1c2
 github.com/sirupsen/logrus.(*Entry).Log(0xc001d64150, 0x4, 0xc001fb1088, 0x1, 0x1)
    /go/pkg/mod/github.com/sirupsen/logrus@v1.6.0/entry.go:287 +0xeb
 github.com/sirupsen/logrus.(*Entry).Info(...)
    /go/pkg/mod/github.com/sirupsen/logrus@v1.6.0/entry.go:304
 github.com/argoproj/argo-cd/util/grpc.logRequest(0xc001d64070, 0xc000058dc0, 0x3a, 0x210a940, 0xc001fa8000, 0x267fd00, 0xc000e2a4b0, 0x1)
    /go/src/github.com/argoproj/argo-cd/util/grpc/logging.go:26 +0x106
 github.com/argoproj/argo-cd/util/grpc.PayloadUnaryServerInterceptor.func1(0x267fd00, 0xc000e2a4b0, 0x210a940, 0xc001fa8000, 0xc001e8c480, 0xc000b980a0, 0x221dadb, 0xc0013d8400, 0xc0013d8400, 0x1e5ab20)
    /go/src/github.com/argoproj/argo-cd/util/grpc/logging.go:77 +0x196
 github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x267fd00, 0xc000e2a4b0, 0x210a940, 0xc001fa8000, 0x2063d60, 0xc000e2a480, 0x267fd00, 0xc000e2a4b0)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x9c
 github.com/argoproj/argo-cd/util/grpc.UserAgentUnaryServerInterceptor.func1(0x267fd00, 0xc000e2a4b0, 0x210a940, 0xc001fa8000, 0xc001e8c480, 0xc000b980a0, 0xc0001fb080, 0x267fd00, 0xc000e2a4b0, 0x0)
    /go/src/github.com/argoproj/argo-cd/util/grpc/useragent.go:22 +0xa2
 github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x267fd00, 0xc000e2a4b0, 0x210a940, 0xc001fa8000, 0x0, 0x0, 0x263de00, 0xc00081b680)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x9c
 github.com/grpc-ecosystem/go-grpc-middleware/auth.UnaryServerInterceptor.func1(0x267fd00, 0xc0001fb080, 0x210a940, 0xc001fa8000, 0xc001e8c480, 0xc000b980a0, 0xc0001fb0e0, 0x3, 0x3, 0x7fa473ed4d20)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/auth/auth.go:46 +0x108
 github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x267fd00, 0xc0001fb080, 0x210a940, 0xc001fa8000, 0x26, 0xc000b980f0, 0x0, 0x0)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x9c
 github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1(0x267fd00, 0xc0001fb080, 0x210a940, 0xc001fa8000, 0xc001e8c480, 0xc000b980a0, 0x0, 0x0, 0x0, 0x0)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0xad
 github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x267fd00, 0xc0001fb080, 0x210a940, 0xc001fa8000, 0x26, 0xbfda271fa2f5bb92, 0x3976644b65, 0x3a41be0)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x9c
 github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus.UnaryServerInterceptor.func1(0x267fd00, 0xc0001faf90, 0x210a940, 0xc001fa8000, 0xc001e8c480, 0xc000b980a0, 0x50, 0x48, 0x20b10c0, 0x0)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/logging/logrus/server_interceptors.go:32 +0x110
 github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x267fd00, 0xc0001faf90, 0x210a940, 0xc001fa8000, 0x7fa49ab9d7d0, 0x0, 0xc0008099f8, 0x40c4b6)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:34 +0x9c
 github.com/argoproj/argo-cd/server.bug21955WorkaroundInterceptor(0x267fd00, 0xc0001faf90, 0x210a940, 0xc001fa8000, 0xc001e8c480, 0xc000b980a0, 0xc0001faf90, 0xc000f33700, 0xc000809a28, 0x40cd28)
    /go/src/github.com/argoproj/argo-cd/server/server.go:1014 +0x15b
 github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1(0x267fd00, 0xc0001faf90, 0x210a940, 0xc001fa8000, 0xc001e8c480, 0xc001e8c4a0, 0xc0014a3a98, 0x4f5cb8, 0x207ad20, 0xc0001faf90)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:39 +0x155
 github.com/argoproj/argo-cd/pkg/apiclient/application._ApplicationService_Create_Handler(0x219c2e0, 0xc00013da20, 0x267fd00, 0xc0001faf90, 0xc001fa6000, 0xc000994210, 0x267fd00, 0xc0001faf90, 0xc0014a3b18, 0x455549)
    /go/src/github.com/argoproj/argo-cd/pkg/apiclient/application/application.pb.go:2591 +0x14b
 google.golang.org/grpc.(*Server).processUnaryRPC(0xc0008e4600, 0x269f560, 0xc00021e870, 0xc00015d800, 0xc000994630, 0x3a20dd0, 0x0, 0x0, 0x0)
    /go/pkg/mod/google.golang.org/grpc@v1.15.0/server.go:982 +0x4bd
 google.golang.org/grpc.(*Server).handleStream(0xc0008e4600, 0x269f560, 0xc00021e870, 0xc00015d800, 0x0)
    /go/pkg/mod/google.golang.org/grpc@v1.15.0/server.go:1208 +0x11f0
 google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0013e9f50, 0xc0008e4600, 0x269f560, 0xc00021e870, 0xc00015d800)
    /go/pkg/mod/google.golang.org/grpc@v1.15.0/server.go:686 +0xa1
 created by google.golang.org/grpc.(*Server).serveStreams.func1
    /go/pkg/mod/google.golang.org/grpc@v1.15.0/server.go:684 +0xa1
```

**Comment by jannfis** [comment]
I think this might be related to logging in JSON format, therefore this issue might be a duplicate of #4117. It has been fixed, but not released yet. 

@alexmt we might want to cherry-pick #4319 into 1.7 branch, what do you think?

**Comment by jessesuen** [comment]
Thanks for triage @jannfis . we'll cherry pick this into 1.7

**Comment by alexmt** [comment]
Cherry-picked the fix into 1.7
